### PR TITLE
[KAIZEN] Clarify mpt, vm and ledger packages

### PR DESCRIPTION
**Pull request by jb-lunatech**
# Description

- Move `NodeKeyValueStorage` to `storage.db`
- Remove unused dependency on `InMemoryWorldStateStorage` in `Blockchain`
- Improve readability with `final` and `private` modifiers
- Add scaladoc, comments and typos

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/mpt/MerklePatriciaTrie.scala [hunk 1]
```diff
@@ -167,15 +162,15 @@ class MerklePatriciaTrie[K, V] private (private[mpt] val rootNode: Option[MptNod
...
+  /** This function inserts a (key-value) pair into the trie. If the key is already associated with another value it is updated.
     *
     * @param key
     * @param value
     * @return New trie with the (key-value) pair inserted.
     * @throws io.iohk.ethereum.mpt.MerklePatriciaTrie.MPTException if there is any inconsistency in how the trie is build.
     */
   override def put(key: K, value: V): MerklePatriciaTrie[K, V] = {
-    val keyNibbles = HexPrefix.bytesToNibbles(kSerializer.toBytes(key))
+    val keyNibbles = mkKeyNibbles(key)
```
Comment (leo-bogastry) [inline src/main/scala/io/iohk/ethereum/mpt/MerklePatriciaTrie.scala:173]: The PR is looking good, I see you tidied things up 👍 
My only question, just for my understanding, is what is the difference between these 2 lines? Does it change something?

Comment (jb-lunatech) [reply src/main/scala/io/iohk/ethereum/mpt/MerklePatriciaTrie.scala:173]: https://github.com/input-output-hk/mantis/pull/1112/files/7c707aeb564b728180caa52be3685208e2695cc6#diff-12a76375555a1e292f13c2b0fd5ae9ab995a4797aa4ca19c41e7f33b63813ed4R163 `mkKeyNibbles` is a private function that exactly does `HexPrefix.bytesToNibbles(kSerializer.toBytes(key))` :)

Comment (leo-bogastry) [reply src/main/scala/io/iohk/ethereum/mpt/MerklePatriciaTrie.scala:173]: Oh I guess I missed that :)

**Comment by jb-lunatech** [comment]
Closing this outdated PR

**Pull request closed without merging**